### PR TITLE
walletrpc: add default values to listunspent

### PR DESCRIPTION
**Pull request by ErikEk**
Adds default values to walletrpc ListUnspent RPC call as requested in #6028.

Fixes https://github.com/lightningnetwork/lnd/issues/6028

### CODE REVIEW

**Review by positiveblue — APPROVED** [review]
Thank you for the changes and for taking the time to make the `ListUnspent` comment clear.

File: lnrpc/walletrpc/walletkit.proto [hunk 1]
```diff
@@ -240,6 +246,14 @@ message ListUnspentRequest {
 
     // An optional filter to only include outputs belonging to an account.
     string account = 3;
+
+    /*
+    When min_confs and max_confs are zero, setting false implicitly
+    overrides max_confs	to be MaxInt32, otherwise max_confs remains
```
[the remark this answers is not in the document]
Comment (ErikEk) [reply lnrpc/walletrpc/walletkit.proto]: @positiveblue I'm not sure if I saw the problem. Hope this look better.

**Review by ellemouton — COMMENTED** [review]
Thanks for adding these @ErikEk ! since this is kinda repeating what is done on the lncli side, i think we can remove the default values from lncli now and just pass through the already existing unconfirmed_only flag in the request. Just so that we dont have duplicated logic for this. wdyt?

**Review by ellemouton — APPROVED** [review]
tACK! 🚀 thanks for this @ErikEk !

**Comment by ellemouton** [comment]
> As is, there is no rpc (only REST) endpoint for listunspent functionality in the walletrpc subserver, hence the will be no duplication of logic.

Indeed, you are correct. My bad 👍

**Comment by joostjager** [comment]
We found out that this PR is actually a breaking change for users that user min and max both at zero.

**Comment by ErikEk** [comment]
@joostjager Let me look i to it.

**Comment by ErikEk** [comment]
@joostjager We had to override max_confs to be a MaxInt32, in order to return all confirmed and unconfirmed utxos as a default response. Before the change the command didn't worked as excepted.

**Comment by ErikEk** [comment]
I think this merge broke '/v1/utxos' but it was depreciated already.

**Comment by joostjager** [comment]
>We had to override max_confs to be a MaxInt32, in order to return all confirmed and unconfirmed utxos as a default response. Before the change the command didn't worked as excepted.

Yes, so this is exactly the breaking change. Previously the default behavior was to return only the unconfirmed utxos, and after the change it returns everything. Users relying on that can get into trouble.

**Comment by ErikEk** [comment]
@joostjager At this moment I'm not sure the best decision would be to reverse it to the old behavior, since we get the same issue but reversed (since this has been in the client for some time).

**Comment by joostjager** [comment]
Yes, that's a dilemma indeed. Given that nobody complained yet beside me since the release of 0.15, it's probably fine to leave as is. Mainly wanted to flag it to perhaps give extra attention to this for future changes.

**Pull request merged**